### ISSUE TITLE
Panic running with `--rust` and `--everything`

### ISSUE THREAD

**Issue by doprz** [pacak/cargo-show-asm#155]
Issue:
Running `cargo asm --rust --everything` panics and outputs this error:
`thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 210'`

stack backtrace:
```
0: rust_begin_unwind
1: core::panicking::panic_fmt
2: core::panicking::panic_bounds_check
3: cargo_show_asm::asm::dump_range
4: cargo_show_asm::asm::dump_function
5: cargo_asm::main
```

**Comment by pacak** [comment]
Seems to be working for me. 

- What platform are you on?
- Do you have rust sources installed?
- What is the version of `rustc` you are using?
- What is the version of `cargo-show-asm` you are using?
- Does it work for a separate functions, without `--everything`?

**Comment by doprz** [comment]
I'm on Linux x64
rustc 1.68.1
cargo-show-asm 0.2.15

Yes, it works fine if I only use `cargo asm --everything ` but panics when running `cargo asm --rust --everything `
I also managed to trace the error to this section of `cargo-show-asm/asm.rs:167`:

```rust
if let Some((fname, file)) = files.get(&loc.file) {
    let rust_line = &file[loc.line as usize - 1];
    let pos = format!("\t\t// {} : {}", fname.display(), loc.line);
    safeprintln!("{}", color!(pos, OwoColorize::cyan));
    safeprintln!(
        "\t\t{}",
        color!(rust_line.trim_start(), OwoColorize::bright_red)
    );
}
```

I'm running this command on v1.0.2 of `rs-cube` https://github.com/doprz/rs-cube/tree/v1.0.2 if you'd like to test it out.

**Comment by pacak** [comment]
> Yes, it works fine if I only use cargo asm --everything but panics when running cargo asm --rust --everything 

I meant if let's say `cargo asm rs_cube::main --rust` works, but `cargo asm --rust --everything` fails...

Either way it works for me for `rs_cube` as well, I'm not 1.68.0 though but should be similar.

Can you try running `cargo asm rs_cube::main --rust -vvv` and either pasting it here or looking for suspicious bits??

**Comment by doprz** [comment]
Interesting, I just tried running the same thing on macOS and it reports a bunch of missing files however I'm able to compile just fine.

Errors (example):
`File not found /private/tmp/rust-20230325-7327-rbrpyq/rustc-1.68.1-src/library/core/src/option.rs`

**Comment by doprz** [comment]
Everything is default and I just updated via `rustup update` and got `info: cleaning up downloads & tmp directories` to make sure it isn't a cache issue.

**Comment by pacak** [comment]
Magic. Anyway, I pushed a branch `strange` that would try to use your folder structure too. Can you check if it helps?

**Comment by doprz** [comment]
Thank you for such a quick reply. That worked like magic.
It seems that macOS creates a tmp dir at `/private/tmp/` which caused `cargo-show-asm` to be unable to find it.

**Comment by doprz** [comment]
Also thank you for creating such an amazing piece of software. It's been very helpful in the development of `rs-cube` so I'd like to add your name to the next version's release notes if you're ok with that to credit you for your contribution.

**Comment by pacak** [comment]
macOS? I thought you said you are on Linux.

`/private/tmp` is not a real path, I strip that part and replace it with a sysroot path as `rustc` reports it.

If it works fine for you - I'll make a release soon-ish.

**Comment by pacak** [comment]
> so I'd like to add your name to the next version's release notes if you're ok with that to credit you for your contribution.

Not sure I deserve it, but no objections :)

**Comment by doprz** [comment]
Yes, sorry for the confusion. I'm testing on both Linux and macOS.
Ok, that makes sense. I misread that.
Looking forward to that release. If you need me to test out something for both Linux or macOS before releasing lmk.

**Comment by pacak** [comment]
Just to confirm, branch `strange` works for you both on Linux and macOS?

**Comment by doprz** [comment]
That is correct.

**Comment by pacak** [comment]
Released 0.2.16

**Comment by doprz** [comment]
> Released 0.2.16

Just tested it out, and everything works as expected. Thank you!👌
